src/odata-to-abstract-sql.ts

```diff
@@ -823,56 +823,87 @@ export class OData2AbstractSQL {
 					}
 				}
 			}
-			if (isBindReference(key)) {
-				const bind = this.Bind(key);
-				const referencedField: ReferencedFieldNode = [
-					'ReferencedField',
-					resource.tableAlias,
-					resource.idField,
-				];
-				return [comparison.eq, referencedField, bind];
+			return this.BaseKey(resource, key);
+		}
+	}
+	BaseKey(
```

Comment:
Use "Hide whitespace" to review this, and then most of this function will look like a move, since BaseKey is mostly extracted from the PathKey.